speedviewlib/src/main/java/com/github/anastr/speedviewlib/RaySpeedometer.kt

```diff
@@ -139,27 +149,35 @@ open class RaySpeedometer @JvmOverloads constructor(
         updateBackgroundBitmap()
     }
 
+    open fun getRayStartDegree(): Float {
+        return getDegreeAtSpeed(rayStartSpeed ?: minSpeed)
+    }
+
     override fun onDraw(canvas: Canvas) {
         super.onDraw(canvas)
 
         canvas.save()
         canvas.rotate(getStartDegree() + 90f, size * .5f, size * .5f)
-        var i = getStartDegree()
-        while (i < getEndDegree()) {
-            if (degree <= i) {
-                rayMarkPaint.color = markColor
-                canvas.drawPath(markPath, rayMarkPaint)
-                canvas.rotate(degreeBetweenMark.toFloat(), size * .5f, size * .5f)
-                i += degreeBetweenMark
-                continue
+
+        val zeroDegree = getRayStartDegree()
+        val rangeCheck = if (degree <= zeroDegree) {
+            degree.toInt()..zeroDegree.toInt()
+        } else {
+            zeroDegree.toInt()..degree.toInt()
+        }
+
+        for (i in getStartDegree()..getEndDegree() step degreeBetweenMark) {
+            val drawWith = if (!rangeCheck.contains(i)) {
```

Comment:
This part I tried to clean up. These lines seemed to be present in both parts of the conditional

```
canvas.drawPath(markPath, rayMarkPaint)
canvas.rotate(degreeBetweenMark.toFloat(), size * .5f, size * .5f)
i += degreeBetweenMark
```
The only thing that really looked to be changing was the paint.